layers/+spacemacs/spacemacs-layouts/packages.el

```diff
@@ -209,7 +209,11 @@
       (spacemacs|define-custom-layout "@Spacemacs"
         :binding "e"
         :body
-        (spacemacs/find-dotfile)))
+        (spacemacs/find-dotfile))
+      (when (and spacemacs-persp-unique-shell
+                 (configuration-layer/package-usedp 'shell-pop))
+        (advice-add 'persp-switch :after #'set-persp-specific-shell)
```

Comment:
When I tried running this locally I kept getting an error that the `set-persp-specific-shell` definition is void, so I updated this to be `spacemacs//set-persp-specific-shell` and then it worked like a charm 👌 . This is a life saver by the way! Been looking for something like this for awhile... well done.

Reply:
@jonyamo fwiw, I do have that function. I'm on `develop` though, maybe you're on `master`?

Reply:
@aaronjensen I'm on `develop` as well, but don't see that function defined anywhere... perhaps I have something amiss with my project. Would you mind pointing me to its definition?

Reply:
err, my bad. It's defined in this patch and it was changed. It looks like you're right, this line does need to change to `spacemacs//set-persp-specific-shell` cc @Ps-Pencil